File: #1-25 (C, Java, Python)/Java/Euler4.java
//Euler4 ~ Largest palindrome product
import java.util.concurrent.TimeUnit;
import java.lang.*;

public class Euler4
{
    public static void main(String[] args) throws InterruptedException
	{
		long startTime = System.nanoTime();
		
		String result = "0";
		int largest = 0;
		int length  = 0;
		int k       = 0;
		for(int i = 0; i <= 999; i++)
		{
			for(int j = 0; j <= 999; j++)
			{
				k             = i*j;
				result = Integer.toString(k);
				length        = result.length();
				
				if(length%2 == 0)
				{
					String s1 = result.substring(0,(length/2));
					String s2 = result.substring((length/2),length);
					s2 = new StringBuilder(s2).reverse().toString();
					
					if(s1.equals(s2))
					{
						if(k > largest)
						{
							largest = k;
						}
					}
					
				}
				else
				{
					if(length != 1)
					{
						String s1 = result.substring(0,(length/2));
						String s2 = result.substring((length/2+1),length);
						s2 = new StringBuilder(s2).reverse().toString();
						
						if(s1.equals(s2))
						{
							if(k > largest)
							{
								largest = k;
							}
						}
					}
				}
			}
		}

		long endTime = System.nanoTime();
		long timeElapsed = endTime - startTime;
		System.out.println("Solution: " + largest + "\n");
		System.out.println("Program completed in " + timeElapsed/1000000000 + " seconds.");
    }
}

File: #1-25 (C, Java, Python)/Python/Euler4.py
#Euler4 ~ Largest palindrome product
import timeit
start = timeit.default_timer()

result  = ""
largest = 0
length  = 0
k       = 0
for i in range(1,1000):
    for j in range(1,1000):
        k = i*j
        result = str(k)
        length = len(result)
        
        if((length%2) == 0):
            s1 = result[0:length//2]
            s2 = result[length//2:length]
            s2 = s2[::-1]
            
            if(s1 == s2):
                if(k > largest):
                    largest = k
        else :
            if(length!= 1):
                s1 = result[0:length//2]
                s2 = result[(length//2+1):length]
                s2 = s2[::-1]
                
                if(s1 == s2):
                    if(k > largest):
                        largest = k
        
        
stop = timeit.default_timer()
print("Solution: ",largest,"\n")
print("Program completed in","{0:.3f}".format(stop - start), "seconds.")